packages/animations/test/shared_axis_transition_test.dart

```diff
@@ -160,6 +160,63 @@ void main() {
       },
     );
 
+    testWidgets(
+      'Child render count between transitions',
+      (WidgetTester tester) async {
+        final GlobalKey<NavigatorState> navigator = GlobalKey<NavigatorState>();
+        const String bottomRoute = '/';
+        const String topRoute = '/a';
+
+        await tester.pumpWidget(
+          _TestWidget(
+            navigatorKey: navigator,
+            contentBuilder: (RouteSettings settings) {
+              return _StatefulTestWidget(
+                key: ValueKey<String>(settings.name),
+                name: settings.name,
+              );
+            },
+            transitionType: SharedAxisTransitionType.horizontal,
+          ),
+        );
+
+        final _StatefulTestWidgetState bottomState = tester.state(
+          find.byKey(const ValueKey<String>(bottomRoute)),
+        );
+
+        // builds once on start
+        expect(bottomState.rebuildCount, 1);
```

Comment:
Checking how often build was called is not appropriate. The framework is free to call build as often as it wants. Instead, you should check that the state object associated with the widget hasn't changed after the transition has started.